common4j/src/main/com/microsoft/identity/common/java/providers/nativeauth/NativeAuthRequestProvider.kt

```diff
@@ -55,6 +63,61 @@ class NativeAuthRequestProvider(private val config: NativeAuthOAuth2Configuratio
     private val resetPasswordSubmitEndpoint = config.getResetPasswordSubmitEndpoint().toString()
     private val resetPasswordPollCompletionEndpoint = config.getResetPasswordPollCompletionEndpoint().toString()
 
+    //region /signup/start
+    internal fun createSignUpStartRequest(
+        commandParameters: SignUpStartCommandParameters
+    ): SignUpStartRequest {
+        LogSession.logMethodCall(TAG, "${TAG}.createSignUpStartRequest")
+
+        return SignUpStartRequest.create(
+            username = commandParameters.username,
+            attributes = commandParameters.userAttributes,
+            challengeType = config.challengeType,
+            clientId = config.clientId,
+            requestUrl = signUpStartEndpoint,
+            headers = getRequestHeaders()
+        )
+    }
+
+    internal fun createSignUpUsingPasswordStartRequest(
+        commandParameters: SignUpStartUsingPasswordCommandParameters
+    ): SignUpStartRequest {
+        LogSession.logMethodCall(TAG, "${TAG}.createSignUpUsingPasswordStartRequest")
+
+        if (commandParameters.password.isEmpty() || commandParameters.password.all { it.isWhitespace() })
+        {
+            var msg = "password can't be empty or consists solely of whitespace characters"
+            throw ClientException("$TAG $msg", msg)
+        }
+
+        return SignUpStartRequest.create(
+            username = commandParameters.username,
+            password = commandParameters.password,
+            attributes = commandParameters.userAttributes,
+            challengeType = config.challengeType,
+            clientId = config.clientId,
+            requestUrl = signUpStartEndpoint,
+            headers = getRequestHeaders()
+        )
+    }
+    //endregion
+
+    //region /signup/challenge
+    internal fun createSignUpChallengeRequest(
+        signUpToken: String
+    ): SignUpChallengeRequest {
+        LogSession.logMethodCall(TAG, "${TAG}.createSignUpChallengeRequest")
+
+        return SignUpChallengeRequest.create(
+            signUpToken = signUpToken,
+            clientId = config.clientId,
+            challengeType = config.challengeType,
+            requestUrl = signUpChallengeEndpoint,
+            headers = getRequestHeaders()
+        )
+    }
+    //endregion
+
```

Comment:
Nit: let's move these down so that all signup methods are sequential in the file, makes it more readable

Reply:
Done